var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -183,6 +183,13 @@ class PyNumpy(PythonPackage):
         depends_on("py-setuptools@:63", when="@:1.25")
         depends_on("py-setuptools@:59", when="@:1.22.1")
 
+    # https://github.com/spack/spack/issues/49983
+    patch(
+        "https://github.com/numpy/numpy/pull/28713.patch?full_index=1",
```

Comment:
It'd be nice if we could grab the merge commit here instead of the PR patch. We shouldn't get into the habit of accepting patches from PRs in the case that people use them to backdoor packages.

Reply:
The checksum will prevent us from grabbing a PR that's been modified post acceptance into Spack. 

However, I guess I'd rather just not normalize patches from PRs in general in the case maintainers get sloppy about checking that they're actually merged / on the path to merge and end up accepting something that is malicious / doesn't get accepted upstream. It'd be nice to write a CI check one day to prevent that.

 Totally understand that the patch for a commit is an awfully long url.
```
https://github.com/numpy/numpy/commit/7771624a4a4c662f936e07bbf74dd7d553225f23.patch
```

it looks like the following works, but may also be a little less safe for collisions later on one day.
```
https://github.com/numpy/numpy/commit/7771624.patch
```

Reply:
I'm fine with changing from PR to commit patches, but we should first agree on this and add a test for it. We already have tests to ensure `?full_index=1` is added, it should be easy to add additional tests to avoid PR patches.

Reply:
I'll bring it up to the team at our next meeting. I agree this should be pretty easy to add to the tests :)

Reply:
It looks like the audit check is already supposed to be checking for pull request patches and failing on them. Obviously that logic isn't working correctly...

Reply:
@tldahlgren this might be of interest to you. I'm willing to give fixing the bug in the audit checks a go, but sounds like we need more testing around the audit checks.

Reply:
After digging into this more, it looks like the existing checks prevent commits from PRs, but seem to intentionally allow the patch from a full PR. I might argue for a change to this policy / audit check, but I see no reason to hold up this PR on that work.